packages/core/src/application.ts

```diff
@@ -88,6 +91,9 @@ export class Application extends Context {
     response: ServerResponse,
   ) {
     this._setupHandlerIfNeeded();
+    if (request.method === 'GET' && request.url === '/openapi.json') {
+      return this._serveOpenApiSpec(request, response);
```

Comment:
+1 for landing this and adding the following issues for post beta:

 - support configurable openapi URLs
 - support yaml / json versions of `/openapi`